ApiClient/Source/Pages/Queries/SinglePageQuery.swift

```diff
@@ -26,27 +26,34 @@
 import Foundation
 import Alamofire
 
-final class SinglePageQuery: Query {
-    typealias Result = JSONPage?
+final class SinglePageQuery: Query {    
+    typealias Result = JSONPage
     
-    let resource: String
+    let url: URL
+    let host: String = "https://api.schedjoules.com"
     let method: HTTPMethod = .get
     let parameters: Parameters = [:]
     let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
     
-    required init(resource: String) {
-        self.resource = resource
+    private init(pageID: String, queryItems: [URLQueryItem]) {
+        // Initialize url components from a string
+        var urlComponents = URLComponents(string: "https://api.schedjoules.com/pages/\(pageID)")
+        // Add query items to the url
+        urlComponents!.queryItems = queryItems
+        // Set the url property to the url constructed from the components
+        self.url = urlComponents!.url!
     }
     
-    /// Initialize with a given Page ID and automatically add locale and
-    /// location parameter to the pages URL
+    /// Initialize with a given Page ID and automatically add locale parameter
     convenience init(pageID: String) {
-        self.init(pageID: pageID, locale: Locale.preferredLanguages[0].components(separatedBy: "-")[0])
+        self.init(pageID: pageID,
```

Comment:
why don't you just delegate to the other ctor which takes a `locale` argument? Avoid code duplication.